pkg/router/f5/plugin.go

```diff
@@ -259,6 +259,11 @@ func (p *F5Plugin) HandleEndpoints(eventType watch.EventType,
 
 			glog.V(4).Infof("Deleting pool %s", poolname)
 
+			// Note: deletePool will throw errors if the route
+			//       has not been deleted as the policy would
+			//       still refer to the pool. That is ok as the
+			//       pool will still get deleted when the route
+			//       gets deleted.
```

Comment:
The problem is on the last endpoint going away, we also try and delete the pool which would fail because there is a policy associated to the pool - since I saw this happening and an error in the logs. So added a comment in here to indicate that is expected behavior. I think we should still return the error (and log it - which gets done by the caller) as it could well be some other delete failure.